File: sdk/typescript/examples/basic.ts
import { createPublicClient, createWalletClient, http } from "viem";
import { privateKeyToAccount } from "viem/accounts";
import { base } from "viem/chains";
import { EvixClient } from "@inductiv/evix-sdk";

const account = privateKeyToAccount("0x...");

const publicClient = createPublicClient({
  chain: base,
  transport: http(),
});

const walletClient = createWalletClient({
  account,
  chain: base,
  transport: http(),
});

const evix = new EvixClient({
  chainId: 8453,
  contractAddress: "0xYourEvixDeployment",
  publicClient,
  walletClient,
});

async function main() {
  // Simulate first
  const expectedOut = await evix.simulateSwap({
    tokenIn: "0xTokenIn",
    tokenOut: "0xTokenOut",
    amountIn: 1_000_000n,
    deadline: BigInt(Math.floor(Date.now() / 1000) + 60),
    minAmountOut: 0n,
    recipient: account.address,
  });

  console.log("Expected output:", expectedOut);

  // Execute with slippage protection
  const { amountOut, transactionHash } = await evix.swap({
    tokenIn: "0xTokenIn",
    tokenOut: "0xTokenOut",
    amountIn: 1_000_000n,
    deadline: BigInt(Math.floor(Date.now() / 1000) + 60),
    minAmountOut: (expectedOut * 995n) / 1000n, // 0.5% slippage
    recipient: account.address,
  });

  console.log("Swap executed:", { amountOut, transactionHash });
}

main().catch(console.error);

File: sdk/rust/examples/basic.rs
use alloy::primitives::{address, U256};
use evix_sdk::{EvixClient, SwapParams};

#[tokio::main]
async fn main() {
    let client = EvixClient::new(
        8453,
        address!("0x1234567890123456789012345678901234567890"),
    );

    let params = SwapParams {
        token_in: address!("0x1111111111111111111111111111111111111111"),
        token_out: address!("0x2222222222222222222222222222222222222222"),
        amount_in: U256::from(1_000_000u64),
        deadline: U256::from(1700000000u64),
        min_amount_out: U256::from(995_000u64),
        recipient: address!("0x3333333333333333333333333333333333333333"),
    };

    // Encode calldata (works without a provider)
    let calldata = client.encode_swap(&params);
    println!("Encoded calldata length: {}", calldata.len());

    // Swap requires a configured provider
    match client.swap(params).await {
        Ok(result) => println!("Swap result: {:?}", result),
        Err(e) => println!("Expected error (no provider): {}", e),
    }
}
